### PR TITLE
perf(RHINENG-9788): Remove unused indexes

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address [RHINENG-9788](https://issues.redhat.com/browse/RHINENG-9788). Removes 4 unused indexes (hosts_subscription_manager_id_index, idx_reporter, idxdisplay_name, idxsap_sids).

There were 2 more considered for deletion that I'm not removing yet:
- idxinsightsid: Unused in Stage, but is used in Prod.
- idx_operating_system: This isn't used in either env, but we still need an index for this. I'll work with David Halasz to figure out why it's not used and determine whether we can update a WHERE or if we have to replace the index.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: migrations/versions/90879eb18c66_remove_unused_indexes.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+def downgrade():
+    with op.get_context().autocommit_block():
+        op.create_index(
+            "hosts_subscription_manager_id_index",
+            "hosts",
+            [sa.text("(canonical_facts ->> 'subscription_manager_id')")],
+        )
+        op.create_index("idx_reporter", "hosts", ["reporter"])
```
Comment (chambridge) [inline migrations/versions/90879eb18c66_remove_unused_indexes.py:35]:
```suggestion
        op.create_index(
            "hosts_subscription_manager_id_index",
            "hosts",
            [sa.text("(canonical_facts ->> 'subscription_manager_id')")],
            postgresql_concurrently=True,
            if_not_exists=True,
        )
        op.create_index(
            "idx_reporter", 
            "hosts", 
            ["reporter"],
            postgresql_concurrently=True,
            if_not_exists=True,
        )
```

Shouldn't these both have the if not exists and concurrently options?

**Review by chambridge — CHANGES_REQUESTED** [review]
quick comment on the downgrade.

File: migrations/versions/90879eb18c66_remove_unused_indexes.py (hunk 1 above)
Comment (kruai) [reply migrations/versions/90879eb18c66_remove_unused_indexes.py:35]: Yep! Copy/paste error

**Review by chambridge — APPROVED** [review]
LGTM

**Pull request merged**